suite-common/formatters/src/FormatterProvider.tsx

```diff
@@ -32,43 +34,45 @@ export type Formatters = {
         string,
         CryptoAmountFormatterDataContext
     >;
-    CoinBalanceFormatter: Formatter<string, string>;
-    CurrencySymbolFormatter: Formatter<NetworkSymbol, string>;
+    NetworkSymbolFormatter: Formatter<NetworkSymbol, string>;
     SignValueFormatter: Formatter<SignValue | undefined, string>;
     FiatAmountFormatter: Formatter<
         string | number,
         string | null,
         FiatAmountFormatterDataContext<FormatNumberOptions>
     >;
+    CryptoToFiatAmountFormatter: Formatter<string, string | null>;
 };
 
 export const FormatterProviderContext = createContext<Formatters>({} as Formatters);
 
 export const getFormatters = (config: FormatterConfig): Formatters => {
     const CryptoAmountFormatter = prepareCryptoAmountFormatter(config);
-    const CoinBalanceFormatter = prepareCoinBalanceFormatter(config);
-    const CurrencySymbolFormatter = prepareCurrencySymbolFormatter(config);
+    const NetworkSymbolFormatter = prepareNetworkSymbolFormatter(config);
     const FiatAmountFormatter = prepareFiatAmountFormatter(config);
+    const CryptoToFiatAmountFormatter = prepareCryptoToFiatAmountFormatter(config);
 
     return {
         CryptoAmountFormatter,
-        CoinBalanceFormatter,
-        CurrencySymbolFormatter,
+        NetworkSymbolFormatter,
         FiatAmountFormatter,
+        CryptoToFiatAmountFormatter,
         SignValueFormatter,
     };
 };
 
 export const FormatterProvider = ({ config, children }: FormatterProviderProps) => {
     const intl = useIntl();
+    const coins = useSelector(selectCoins);
```

Comment:
Please move this to formatter itself, because this still going to rerender all formatters if `coins` change which is very often. (I know it's not convetional approach but this will otherwise a lot).

Also please test it afterwards using flash on rerender etc. that only neccessary things are rerendered.